### PR TITLE
[`refurb`] Implement `fstring-number-format` (`FURB116`)

### PR DESCRIPTION
**Pull request by diceroll123**
## Summary

Adds `FURB116`

See #1348 

## Test Plan

`cargo test`

### CODE REVIEW

**Comment by diceroll123** [comment]
Not too sure how I'd reliably be able to support expressions that can include strings, such as calls, like in my `int(f"{num}")` test, for example. Please advise if that can be better! 😄

File: crates/ruff_linter/src/rules/refurb/rules/fstring_number_format.rs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        return;
+    };
+
+    let Expr::Name(ast::ExprName { id, .. }) = func.as_ref() else {
+        return;
+    };
+
+    if !checker.semantic().is_builtin(id) {
+        return;
+    }
```
Comment (diceroll123) [inline crates/ruff_linter/src/rules/refurb/rules/fstring_number_format.rs:112]: If #10919 is merged before this, I'll update accordingly.

Comment (AlexWaygood) [reply crates/ruff_linter/src/rules/refurb/rules/fstring_number_format.rs:113]: it was merged ;)

**Review by charliermarsh — APPROVED** [review]
Thanks!

**Pull request merged**